### PR TITLE
[miio] Update yeelink.light.light15.json

### PR DESCRIPTION
**Pull request by michiii1337**
changed Datatype from "Number" to "Dimmer" at channel "ambientBrightness" same issue openhab#9936 (but already fixed)

Signed-off-by: Michael Schmidts [michael.schmidts@online.de](mailto:michael.schmidts@online.de)

### CODE REVIEW

**Review by marcelrv — CHANGES_REQUESTED** [review]
Hi @michiii1337 
I recognize the issue with the dimmer vs number. I recall the problem is that I wanted to make it 2 ways, meaning that setting the dimming level in the app is also reflected in the dimming level in OH. As the device reports the level in whole number I believe it did not work well which is why I made it a number.

Can you test on your end and confirm that setting the dimming level via the app is properly reflected in OH.
If not, we prob need to add some calc to multiply/divided by 100... which is nowadays supported by the binding so I expect doable.

Also would be great to check some of the other yeelights if you have a chance for the same matter.
Lastly, once we have confirmed it all working right, please run the readme maker (in the test folder) to update the readme documentation accordingly.

**Comment by michiii1337** [comment]
I tested the dimming level via app and openhab. When the datatype is set to Dimmer, both in Xiaomi Home app and Openhab is the same/correct value. Unfortunately, I have no other Yeelights so I can only test this type.

**Comment by michiii1337** [comment]
the readme is now up to date. Hope I did everything right as this is my first pullreqest.

**Comment by marcelrv** [comment]
I think code wise it is fine.
There are some oddities wrt to the automatic generated readme but they are not caused by you and should be fixed by translating the relevant values and than it fixes itself

There is a commit signoff issue with 2 of the commits.  You can try to 'squash' in eclipse and update the the commit comment to include the signoff-by (like you did in your last commit)
Or if whoever is merging this agrees you can just click accept DCO and it will be fine as well. (I don't have rights to merge the PR)

**Review by marcelrv — APPROVED** [review]

File: bundles/org.openhab.binding.miio/README.md [hunk 1]
```diff
@@ -1963,7 +1963,7 @@ Note, not all the values need to be in the json file, e.g. a subset of the param
 | total-clean-area     | Number:Area          | Clean Record - Total Clean Area          |            |
 | total-clean-time     | Number               | Clean Record - Total Clean Time          |            |
 | total-clean-count    | Number               | Clean Record - Total Clean Count         |            |
-| language             | Number               | Language - Language                      | Value mapping `["0"="English","1"="简体中文","2"="Español","3"="Русский","4"="Italiano","5"="Français","6"="Deutsch","7"="한국어","8"="Polski"]` |
+| language             | Number               | Language - Language                      | Value mapping `["0"="English","1"="ç®€ä½“ä¸­æ–‡","2"="EspaÃ±ol","3"="Ð ÑƒÑ�Ñ�ÐºÐ¸Ð¹","4"="Italiano","5"="FranÃ§ais","6"="Deutsch","7"="í•œêµ­ì–´","8"="Polski"]` |
```
Comment (jlaur) [inline bundles/org.openhab.binding.miio/README.md:1966]: Something seems to have gone wrong here with the non-Western languages?

File: bundles/org.openhab.binding.miio/README.md [hunk 2]
```diff
@@ -2794,7 +2794,7 @@ Note, not all the values need to be in the json file, e.g. a subset of the param
 | mode                 | Number               | Robot Cleaner - Mode                     | Value mapping `["0"="Silent","1"="Basic","2"="Medium","3"="Strong"]` |
 | battery-level        | Number:Dimensionless | Battery - Battery Level                  |            |
 | repeat-state         | Switch               | Viomi Vacuum - Repeat State              |            |
-| remember-state       | Switch               | Viomi Vacuum - Remember State            | Value mapping ["0"="关","1"="开"] |
+| remember-state       | Switch               | Viomi Vacuum - Remember State            | Value mapping ["0"="å…³","1"="å¼€"] |
```
Comment (jlaur) [inline bundles/org.openhab.binding.miio/README.md:2797]: And here.

File: bundles/org.openhab.binding.miio/src/main/resources/OH-INF/i18n/basic.properties [hunk 3]
```diff
@@ -2662,13 +2662,13 @@ option.mijia.vacuum.v2-miot.fault-7 = Side-brush-error
 option.mijia.vacuum.v2-miot.fault-8 = Fan-motor-error
 option.mijia.vacuum.v2-miot.fault-9 = Dustbin-error
 option.mijia.vacuum.v2-miot.language-0 = English
-option.mijia.vacuum.v2-miot.language-1 = 简体中文
-option.mijia.vacuum.v2-miot.language-2 = Español
-option.mijia.vacuum.v2-miot.language-3 = Русский
+option.mijia.vacuum.v2-miot.language-1 = ç®€ä½“ä¸­æ–‡
```
Comment (jlaur) [inline bundles/org.openhab.binding.miio/src/main/resources/OH-INF/i18n/basic.properties:2665]: And here.

File: bundles/org.openhab.binding.miio/src/main/resources/OH-INF/i18n/basic.properties [hunk 4]
```diff
@@ -2662,13 +2662,13 @@ option.mijia.vacuum.v2-miot.fault-7 = Side-brush-error
...
+option.mijia.vacuum.v2-miot.language-2 = EspaÃ±ol
+option.mijia.vacuum.v2-miot.language-3 = Ð ÑƒÑ�Ñ�ÐºÐ¸Ð¹
```
Comment (jlaur) [inline bundles/org.openhab.binding.miio/src/main/resources/OH-INF/i18n/basic.properties:2667]: And here.

File: bundles/org.openhab.binding.miio/src/main/resources/OH-INF/i18n/basic.properties [hunk 5]
```diff
@@ -2662,13 +2662,13 @@ option.mijia.vacuum.v2-miot.fault-7 = Side-brush-error
...
 option.mijia.vacuum.v2-miot.language-4 = Italiano
-option.mijia.vacuum.v2-miot.language-5 = Français
+option.mijia.vacuum.v2-miot.language-5 = FranÃ§ais
```
Comment (jlaur) [inline bundles/org.openhab.binding.miio/src/main/resources/OH-INF/i18n/basic.properties:2669]: And here.

File: bundles/org.openhab.binding.miio/src/main/resources/OH-INF/i18n/basic.properties [hunk 6]
```diff
@@ -2662,13 +2662,13 @@ option.mijia.vacuum.v2-miot.fault-7 = Side-brush-error
...
 option.mijia.vacuum.v2-miot.language-6 = Deutsch
-option.mijia.vacuum.v2-miot.language-7 = 한국어
+option.mijia.vacuum.v2-miot.language-7 = í•œêµ­ì–´
```
Comment (jlaur) [inline bundles/org.openhab.binding.miio/src/main/resources/OH-INF/i18n/basic.properties:2671]: And here.

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks! There seem to be some character set issues that would need to be resolved, and also the DCO check must pass.

**Comment by michiii1337** [comment]
I fixed the characters in the readme. How can i fix the DCO? I can not edit the 3 commits without my sign.

**Comment by marcelrv** [comment]
You can try to 'squash' in eclipse and update the the commit comment to include the signoff-by (like you did in your last commit)

**Comment by michiii1337** [comment]
i tried already to squash the commits but i received a error message: "Unable to squash. Squashing replays all commits up to the last one required for the squash. A merge commit cannot exist among those commits"

**Comment by jlaur** [comment]
> I fixed the characters in the readme. How can i fix the DCO? I can not edit the 3 commits without my sign.

You could try to click "Details" in the DCO check line and follow the steps described, or you could have a look at this:
https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

**Comment by michiii1337** [comment]
i followed the instructions under "Details" and now the DCO failed again... is it easier to start from zero and make an new PR?

**Comment by marcelrv** [comment]
Maybe what's working is to do first a rebase, than I think it will allow you to do the squash.
Making a new branch from current and copy the free changes is indeed also a way

**Comment by michiii1337** [comment]
i have done a rebase and i got the same error message when i try to squash

**Comment by michiii1337** [comment]
ok, now i have done a rebase. before that i seem to have done only a merge. Sorry for the many questions and problem in the PR I am still learning. Hope now the commits can be merged.

**Comment by jlaur** [comment]
> ok, now i have done a rebase

I'm sorry, but that did no go well, as there are now changes from other bindings included. Perhaps at this point it would be best to:
- Backup your changes files.
- Delete your branch locally.
- Create a new branch from main with the **same name** as the deleted branch.
- Apply your changes again (by copying the changed files).
- Force push the new branch.

**Comment by michiii1337** [comment]
my branch is "main". if i delete this branch how should i create new branch from main?

**Comment by jlaur** [comment]
> my branch is "main". if i delete this branch how should i create new branch from main?

Ah, sorry, I didn't see that. Assuming your fork is new and you don't have other branches to keep, for this PR I would adapt the plan to:
- Backup your repository.
- Delete your fork.
- Create a new fork.
- Apply your changes again (by copying the changed files).
- Commit again to main while making sure signoff is correct.
- Force push main.

I don't know if it will work, but it might be worth a shot. After that, I would then suggest to recreate the fork again, and make sure you don't commit anything to main, as it will most likely cause problems for you. Instead you should create a new branch, and use this when creating a PR.

**Comment by Hilbrand** [comment]
It looks like only 1 commit is wrong. You can also just remove that one commit. Use `git rebase -i HEAD~6`. This will open an editor with the list of commits. Replace `pick` with `d` on the commit you want to remove. Than save the file. After that do a push force.

**Comment by michiii1337** [comment]
Thanks @Hilbrand and @jlaur removed the wrong commit.

**Review by jlaur — APPROVED** [review]
Thank you!

**Pull request merged**